core/shared/src/main/scala/fs2/Segment.scala

```diff
@@ -1431,3 +1432,50 @@ object Segment {
     def defer(t: => Unit): Unit = deferred.addLast(() => t)
   }
 }
+
+trait SegmentInstances {
+  implicit def segmentMonoidInstance[A]: Monoid[Segment[A, Unit]] = new Monoid[Segment[A, Unit]]{
```

Comment:
We could define a `Semigroup[Segment[A,R]]` too, which might necessitate instance prioritization (though perhaps not).

Reply:
Could the `Semigroup` be non-implicit?

The implicit monoid allows for `runFoldMonoid` easily on the default case of segment.